lib/tarantool_server.py

```diff
@@ -1270,3 +1276,79 @@ def wait_lsn(self, node_id, lsn):
 
     def get_log(self):
         return TarantoolLog(self.logfile).positioning()
+
+    def get_iproto_port(self, deadline=None):
+        # Check the `box.cfg.listen` option, if it wasn't defined, just return.
+        res = yaml.safe_load(self.admin('box.cfg.listen', silent=True))[0]
+        if res is None:
+            return
+
+        deadline = deadline or 'nil'
+        sleep_time = 0.001
+        timeout = Options().args.server_start_timeout
+
+        version = yaml.safe_load(
+            self.admin("require('tarantool').version", silent=True)
+        )[0]
+        if version.split('.')[0] == '1':
```

Comment:
You can just execute this code if `box.info.listen` gives `nil`. No need explicit version check.

```lua
while <...> do
    local res = box.info.listen
    if res ~= nil then
        return <...>
    end
    -- <..tarantoool < 2.4.1 code..>
end
```


The explicit version check is worse than direct attempts to use a feature you need, at least because it'll work for versions before 2.4.0-103-g2c3acf2b0.

Reply:
We have some tests where a server starts without waiting for the full load. I don't know exactly, but I think there may be a situation when `box.info.listen` will give `nil` (that's why we do it in the loop) and in this case we will execute 'tarantoool < 2.4.1' code that may produce wrong result.

Reply:
Doesn't it mean that `wait=False wait_load=False` now will perform waiting anyway?

Reply:
Doesn't it mean that we should either:

* if waiting is acquired, do it,
* otherwise run a greenlet to poll the server and set the `iproto` attribute (don't forget to finish it when the server is stopped)?

Reply:
> Doesn't it mean that `wait=False wait_load=False` now will perform waiting anyway?

Looks like not. Note, the `wait=False wait_load=False` and `wait=True wait_load=False` cases are identical from the point of view of getting the iproto socket. Look at [this](https://github.com/tarantool/test-run/blob/b59464f17f391ecffd985a2366777bc3cb354eca/lib/tarantool_server.py#L1143-L1152). When `wait=True wait_load=False`, we test only the admin connection and return. When  `wait=True wait_load=True`, `box.info.listen` gives the socket even though `box.info.status` is `loading`.

Reply:
You assume that `box.cfg()` is successful (or will be successful) and so binding a listening socket for iproto is successful. It may be not the case for a `wait_load=False` instance. AFAIR, we have test cases, where a supplementary server is expected to fail loading.

Reply:
I realized that we shouldn't get the iproto port if `wait=False` because we can get the port only when the admin connection can be established and tested (`wait=True` guarantees it). 

Also, I think we shouldn't get the iproto port when `crash_expected=True`. When a server is going to be failed due to some error (for example, when `box.cfg()` is unsuccessful) and we expect this, we have to provide `crash_expected=True`, otherwise the test will fail.

Reply:
I would like to get the iproto port only when `wait=True wait_load=True crash_expected=False`. In this case, we can use the code (without `while` loop) that you proposed in your first comment. It really simplifies the things, but if a server starts with `wait=False` or `wait_load=False`, the iproto socket will be missing for it. But it's not the case for tests where Unix sockets are used (almost all). Anyway, all tests passed when I did testing, and diff and app tests are deprecated in favor of luatest ones and not produced anymore.